ulohy_05/rodne_cislo.py

```diff
@@ -0,0 +1,61 @@
+while True:
+    r_c = input("Jaké je tvé rodné číslo včetně lomítka? ")
+    pred = r_c[0:6] # 6 čísel před lomítkem
+    za = r_c[7:11]  # 4 čísla za lomítkem
+
+    # Ošetření správného formátu
+    if not (pred.isdigit() and za.isdigit() 
+            and len(r_c) == 11 and r_c[6] == "/"):
+        print("Toto není správný formát. Zkus to znovu.")
+        continue
+    cislo = pred + za       # složení rodného čísla bez lomítka
+
+    # Zjištění, zda je rodné číslo správně
+    # Ověření dělitelnosti 11
+    if int(cislo) % 11 != 0:
+       print("Toto asi nebude správně.")
+       continue
+
+    # Pro zjednodušení převod str na int a rozdělení na dvojčíslí
+    rok = int(cislo[0:2])
+    mes = int(cislo[2:4])
+    den = int(cislo[4:6])
+
+    # Pouze pro roky 1985 - 2022
+    if 22 < rok <85:
+        print("Rok tvého narození není v rozmezí 1985 - 2022")
+        continue
+    
+    # Ošetření čísla měsíce a ženského rodného čísla
+    if 50 < mes <=62:
+        mes = mes - 50
+
+    if mes > 12:
+        print("Nemá rok pouze 12 měsíců?")
+        continue
+    
+    if mes == 0:
+        print("Neexistuje měsíc s číslem 0.")
+        continue
```

Comment:
Mohla by to být i jedna podmínka ale v pohodě :)

Reply:
Ano, jedna podmínka to mohla být, ale pak bych nemohla napsat tyto dvě chybové hlášky, které se mi líbily. 😄